### ISSUE TITLE
[Feature Request]: SEM 

### ISSUE THREAD

**Issue by Asadiibrahim** [jasp-stats/jasp-issues#3004]
### Description

  Hello,  I am a researcher in the field of humanities, and I recently started using the JASP software for statistical analyses. I am very pleased and excited about the software and its efficiency. However, I am facing an issue when performing structural equation modeling (SEM) analyses. Specifically, when I try to run an analysis and request an output with a graph showing the standardized estimates of the latent (predictor) variables and the percentage of explained variance, I receive a graph displaying non-standardized errors (even though they appear as standardized in the tables throughout the analysis). Additionally, the graph does not show the percentage of explained variance next to the dependent latent variable (please see the attached graph).  When watching some videos on YouTube, I noticed that some users had an “Advanced” tab where, apparently, it’s possible to request standardized estimates on the graph and the explained variance (as is also possible in the AMOS ). Could you help me solve this issue? I would appreciate a detailed response, and perhaps a specific video demonstrating what needs to be done.  Thank you in advance

### Is your feature request related to a JASP module?

SEM

### Describe the solution you would like

TO SEE THE R SQUARE ON THE GRAPH AS WELL AS THW ESTIMATED STANDARDIZED. SEE THE GRAPH BELOW

### Additional context

[SEM JAST.docx](https://github.com/user-attachments/files/17682425/SEM.JAST.docx)

**Comment by juliuspfadt** [comment]
@Asadiibrahim Thank you for the request. I cannot fulfil everything you are asking for. As far as I know this has never been possible in the JASP path diagram. It's a reasonable request, and I have implemented the necessary code now and it will hopefully be in the next release of JASP. This is what it would look like:
<img width="1392" alt="Screenshot 2024-11-08 at 20 54 55" src="https://github.com/user-attachments/assets/5e4547b6-710b-4560-af24-e67409c4d861">

**Comment by Asadiibrahim** [comment]
thank you for your quick response.
Actually, it must look like in AMOS (with explained variance in reading comprehension and standardized estimation in the latent predictors):
[cid:301f5281-98ac-4461-8432-8b862809be09]
________________________________

**Comment by Asadiibrahim** [comment]
thank you for your quick response.
Actually, it must look like in AMOS (with explained variance in reading comprehension and standardized estimation in the latent predictors) as in the attached file
[SEM amos and jasp.docx](https://github.com/user-attachments/files/17686471/SEM.amos.and.jasp.docx)
:

**Comment by tomtomme** [comment]
We partly rely on the functions that the R packages give us. As Julius said, some things might not be easy/possible to implement. We might never be able to exactly copy everything that is desirable. Sorry

**Comment by juliuspfadt** [comment]
yes. We are using the R-package semPlot which allows either display the raw estimates or the standardized ones. The screenshot I attached has the fully standardized ones.

**Comment by Asadiibrahim** [comment]
Hello again, and thank you for your response. I would like to clarify that my intention is to improve my use of the software that I also teach my students to use for their analyses. Returning to my question and your response, I fully understand that not everything is possible or feasible to implement. However, what motivates me to keep reaching out is that in some of the videos I've seen on YouTube (I am including a link to one of them here), it does appear possible to perform the SEM analysis—exactly the same analysis I am trying to run—with standardized estimations in the resulting graph (path diagram), and with the option to check a box that I don't have in my software version when performing this analysis. 

As you will see in the video, around the third minute, you can see that the commands the presenter used are identical to mine (prediction), and also, as you’ll notice at the bottom of the screen, there is a window that can be opened called “Advanced.” At the ninth minute, you can see how she uses the “Advanced” button, which gives her the option to select the path diagram, as well as to check a box to show only standardized estimations on the graph. At 11:50, the panel with the explained variance percentage is clearly visible, and at 12:20, you can see the graph with the standardized estimations and explained variance. My point is that it seems this can indeed be done, or perhaps I am missing something, or there has been a change in the versions, and it is different in the version I have (0.19.1). I hope this clarifies my question and will help find an effective solution for using the software.

https://www.google.co.il/search?q=STRUCTURAL+EQUATION+MODEL++r2++in++JASP&sca_esv=d094fb0ad0a20eda&biw=1536&bih=703&tbm=vid&sxsrf=ADLYWIL8DagVLjCwNdRgLsVyFzFF7i1Vzw%3A1731145752108&ei=GDAvZ-CiBvrSp84P2v2QsQ4&ved=0ahUKEwjgju-8_M6JAxV66ckDHdo-JOY4HhDh1QMIDQ&uact=5&oq=STRUCTURAL+EQUATION+MODEL++r2++in++JASP&gs_lp=Eg1nd3Mtd2l6LXZpZGVvIidTVFJVQ1RVUkFMIEVRVUFUSU9OIE1PREVMICByMiAgaW4gIEpBU1AyCBAAGIAEGKIEMggQABiABBiiBDIIEAAYgAQYogQyCBAAGIAEGKIEMggQABiABBiiBEjWcFAAWPNscAB4AJABAJgBgAKgAf8SqgEEMi0xMLgBA8gBAPgBAZgCBqAC1guYAwCSBwMyLTagB7Mk&sclient=gws-wiz-video#fpstate=ive&vld=cid:46445b19,vid:lKYUN5E3MOc,st:0
[Hello again.docx](https://github.com/user-attachments/files/17686896/Hello.again.docx)

**Comment by juliuspfadt** [comment]
Sorry for the misunderstanding. I think in a SEM publication it makes sense to present a graph with the standardized estimates; I dont know if one would also include the explained variance in that graph. However, you are right, that the module has changed in recent years. What you posted is a google search link, so I dont know which video you are referring to. Would you mind just posting a link to the video, so I can get to the bottom of this?

**Comment by juliuspfadt** [comment]
And one more thing, with the change I made now, JASP shows the unexplained variance per manifest variable if standardized estimates are chosen for the plot, because that is the standardized residual variance.

**Comment by Asadiibrahim** [comment]
here the link

https://youtu.be/lKYUN5E3MOc?si=b5mz-z66jcUF9GSv
YouTube<https://youtu.be/lKYUN5E3MOc?si=b5mz-z66jcUF9GSv>
Share your videos with friends, family, and the world
youtu.be

________________________________

**Comment by juliuspfadt** [comment]
Ah okay. So you can obtain the explained variance per indicator by going to output options and checking "R-squared". The graph in the video as well as the graph at the moment shows the unexplained variance per outcome, that is, the residual variance.

**Comment by Asadiibrahim** [comment]
What about the option of "advanced" that includes standardized estimations but not exist in my jasp version? Can we fix it?

קבל ‏Outlook עבור Android‏<https://aka.ms/AAb9ysg>
________________________________

**Comment by juliuspfadt** [comment]
I dont know what you mean. In the current released version of JASP, you can already get R-squared and standardised estimates within the "Output options" menu

**Comment by Asadiibrahim** [comment]
YES, BUT NOT IN THE GRAPH
________________________________

**Comment by juliuspfadt** [comment]
I suppose the frustration is mutual. Let me explain how this works: I have changed the underlying code for JASP on my local machine so the standardized estimates also show up in the graph. This is how I produced the screenshot from here https://github.com/jasp-stats/jasp-issues/issues/3004#issuecomment-2465652992. Now this code needs to be reviewed by another JASP team member in order to become a part of JASP, and thus JASP itself will contain the change that so far, I have only made locally. Once someone has approved my code, all future versions of JASP will contain the functionality to have standardized estimates in the graph. But this takes time. The next version of JASP will have this change, but the current does not.
